spotinst/ocean_aks_virtual_node_group_auto_scaling/fields_spotinst_ocean_aks_virtual_node_group_auto_scaling.go

```diff
@@ -106,17 +116,24 @@ func expandAutoScale(data interface{}, nullify bool) (*azure.VirtualNodeGroupAut
 	m := list[0].(map[string]interface{})
 
 	if v, ok := m[string(Headrooms)]; ok {
+
 		headroom, err := expandHeadrooms(v, nullify)
 		if err != nil {
 			return nil, err
 		}
-		if headroom != nil {
+		if len(headroom) > 0 {
 			autoscale.SetHeadrooms(headroom)
 		} else {
-			autoscale.Headrooms = nil
+			autoscale.SetHeadrooms(nil)
```

Comment:
need to make sure it is not a breaking change for users that already have `autoscale_headroom` in their configuration